bin/XMLToSARIF.sh

```diff
@@ -0,0 +1,126 @@
+print_usage() {
+    echo "Usage: XMLToSARIF.sh -i <inputXmlReport> [-o <outputSarifReport>] [-t <toolHomeDir>]"
+    echo ""
+    echo "Options:"
+    echo "  -i, --inputXmlReport      (required) Path to the input XML report."
+    echo "  -o, --outputSarifReport   Path to the output SARIF report."
+    echo "  -t, --toolHomeDir         Path to the tool home directory."
+    echo ""
+}
+
+check_param() {
+    local param_name="$1"
+    local param_value="$2"
+    if [[ -z "$param_value" || "$param_value" == -* ]]; then
+        echo "Error: Invalid value for $param_name."
+        exit 1
+    fi
+}
+
+getJavaPath() {
+    local parasoftToolRootPath="$1"
+
+    if [ ! -d "$parasoftToolRootPath" ]; then
+        echo "$parasoftToolRootPath is not found."
+        exit 1;
+    fi
+
+    local javaFileName="java"
+
+    local javaPaths=("bin" "bin/dottest/Jre_x64/bin" "bin/jre/bin")
+    for path in "${javaPaths[@]}"; do
+        local javaFilePath="$parasoftToolRootPath/$path/$javaFileName"
+        if [ -f "$javaFilePath" ]; then
+            echo "$javaFilePath"
+            return 0
+        fi
+    done
+
+    return 1
+}
+
+# 1.Initialize variables:
+BASE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+WORK_DIR=$(pwd)
+SARIF_XSL_PATH="$BASE_DIR/xsl/sarif.xsl"
+
+parasoft_tool_home=""
+xml_report_path=""
+sarif_report_path=""
+
+# 2.Read the options
+TEMP=$(getopt -u -o i:o:t: --long inputXmlReport:,outputSarifReport:,toolHomeDir: -n 'XMLToSARIF.sh' -- "$@")
```

Comment:
Can we consider implementing it by ourself, becuase we also need to do the same thing on windows. 
And only seupport long-name options now. 
Need to discuss with Yuqian.

Reply:
It seems that there isn't a similar tool in batch scripts that allows for convenient parsing of command-line options, and we need to parse them manually. If that's the case, I think manually parsing in shell scripts as well would make it easier to keep consistent behaviors. Unless you can find a convenient tool for batch scripts, we can reconsider.